anchorecli/cli/utils.py

```diff
@@ -974,6 +892,58 @@ def format_output(config, op, params, payload):
     return(ret)
 
 
+def format_vulnerabilities(payload, params):
+    obuf = ""
+    if 'query_type' not in params or not params['query_type']:
+        # payload will be a list with what is available as a query for the
+        # given image
+        for query in payload:
+            obuf += "%s: available\n" % query
+        return obuf + "\n"
+
+    if params['query_type'] in ['os', 'non-os', 'all']:
+        header = [
+            'Vulnerability ID', 'Package', 'Severity', 'Fix', 'CVE Refs',
+            'Vulnerability URL', 'Type', 'Feed Source', 'Package Path'
+        ]
+        t = plain_column_table(header)
+        for el in payload['vulnerabilities']:
+            nvd_data = el.get('nvd_data')
+            cve_ids = []
+            for nvd_record in nvd_data:
+                _id = nvd_record.get('id')
+                if _id:
+                    cve_ids.append(_id)
+            row = [
+                el['vuln'], el['package'], el['severity'], el['fix'], ','.join(cve_ids),
+                el['url'], el['package_type'], el['feed_group'], el['package_path']
+            ]
+            t.add_row(row)
+        obuf = obuf + t.get_string(sortby='Severity')
+    else:
+        # XXX Investigate if this can be removed: seems like it isn't possible
+        # to get to this situation with a valid query. The supported types are
+        # os, non-os, and all. This code here will attempt to create a table
+        # with the raw payload which is completely unreadable
+        try:
```

Comment:
I was unable to get to this block on the command line. If there aren't any query arguments it defaults to listing what is available, otherwise it wants `os`, `non-os`, or `all`. I couldn't force this section at all except when unit-testing. The output when forcing it via tests is completely unreadable as it tries to fit dictionary output in a table. I would recommend removing it, but not sure if there is a good reason for this to stay here.

Reply:
go for it if it's unreachable